index.html

```diff
@@ -929,31 +1106,423 @@ <h2>ConsumedThing Examples</h2>
+          Return a {{Promise}} |promise:Promise| and execute the next steps
+          <a>in parallel</a>.
+        </li>
+        <li>
+          If |data| is not {{ReadableStream}} or if |dataUsed| is `true`,
+          reject|promise| with {{SyntaxError}} and abort these steps.
+        </li>
+        <li>
+          Let |reader| be the result of
+          <a href="https://fetch.spec.whatwg.org/#concept-get-reader">
+          getting a reader</a> from |data|. If that threw an exception, reject
+          |promise| with that exception and abort these steps.
+        </li>
+        <li>
+          Let |bytes| be the result of <a href=
+          "https://fetch.spec.whatwg.org/#concept-read-all-bytes-from-readablestream">
+          reading all bytes</a> from |data| with |reader|.
+        </li>
+        <li>
+          Set |dataUsed| to `true`.
+        </li>
+        <li>
+          Let |arrayBuffer| be a new {{ArrayBuffer}} whose contents are |bytes|.
+          If that throws, reject |promise| with that exception and abort these steps.
+        </li>
+        <li>
+          Resolve |promise| with |arrayBuffer|.
+        </li>
+      </ol>
+    </section>
+
+    <section><h3>The <dfn>value()</dfn> function</h3>
+      Parses the data returned from the <a>WoT Interaction</a> and returns a
+      value with the type described by the <a>DataSchema</a> of the interaction
+      if that exists, or by the `contentType` of the <a>Form</a> used for the
+      interaction. The method MUST run the following steps:
+      <ol>
+        <li>
+          Return a {{Promise}} |promise:Promise| and execute the next steps
+          <a>in parallel</a>.
+        </li>
+        <li>
+          If the value of the <a>[[\value]]</a> <a>internal slot</a> is not
+          `undefined`, resolve |promise| with that value and abort these steps.
+        </li>
+        <li>
+          If the value of the |data| property is not a {{ReadableStream}} or if
+          |dataUsed| is `true`, or if |form| is `null` or if |schema| or its
+          |type| are `null` or `undefined`,
+          reject |promise| with {{SyntaxError}} and abort these steps.
+        </li>
+        <li>
+          Let |reader| be the result of
+          <a href="https://fetch.spec.whatwg.org/#concept-get-reader">
+          getting a reader</a> from |data|. If that threw an exception, reject
+          |promise| with that exception and abort these steps.
+        </li>
+        <li>
+          Let |bytes| be the result of <a href=
+          "https://fetch.spec.whatwg.org/#concept-read-all-bytes-from-readablestream">
+          reading all bytes</a> from |data| with |reader|.
+        </li>
+        <li>
+          Set |dataUsed| to `true`.
+        </li>
+        <li>
+          If |form|'s |contentType| is not `application/json` and if a mapping is
+          available in the <a>Protocol Bindings</a> from |form|'s |contentType|
+          to [JSON-SCHEMA], transform |bytes| with that mapping.
```

Comment:
I'd check the content type before reading the response. So that the user has still the opportunity to parse the stream if the content-type has no mapping to JSON-Schema

Reply:
Developers can read the `form`'s `contentType` before deciding to call `value()` or use the stream.

BTW I am thinking to add a note that implementations MAY decide to duplicate the stream, i.e. `value()` would consume one, but then `data` would be still available for read. The only thing devs should do is check the `dataUsed` attribute beforehands in order to avoid errors - or then handle the errors.

Reply:
> Developers can read the form's contentType before deciding to call value() or use the stream.

yeah, right. However, I think it does not change the algorithm so much and it gives an opportunity to handle this case only if the method fails. what do you think?

> BTW I am thinking to add a note that implementations MAY decide to duplicate the stream, i.e. value() would consume one, but then data would be still available for read. The only thing devs should do is check the dataUsed attribute beforehands in order to avoid errors - or then handle the errors.

Can you please provide a use case when you want to read the stream anyway even if you **have** already a parsed value? Aside from when the value function fails I cannot find any. Moreover, even in such a scenario, I am not sure if another stream would be valuable. Here it is a list of possible `value` errors (please correct me if I miss any) :
- malformed JSON: any other parser would fail anyway --> no need to read from a stream
- no mapping from the content type to JSONSchema: As I suggested the algorithm of the value function can be engineered to not consume the stream --> no need to have another stream
- invalid data schema: I think this is the only use case to try consuming the stream manually after value failure. However, should we encourage this practice? Invalid data schema means that something is really off...

Reply:
I just want to avoid duplicate network requests, since that is far more expensive than duplicating a stream (which boils down to pointer ownership and no data duplication necessarily).

The normal flow would be to try to use `value()`, which may fail, in which case  using `data` may be attempted.

Now I think I misunderstood your first comment - what you seem to wish is that `value()` fails as early as possible before disturbing the stream. Yes, let's try to do that first and see how it works.